server/game/core/card/Card.ts

```diff
@@ -361,6 +362,10 @@ export class Card extends OngoingEffectSource {
         return !this.facedown && !this.hasRestriction(AbilityRestriction.InitiateKeywords, context);
     }
 
+    protected getGainedAbilityEffects<TAbility>(abilityType: AbilityType): TAbility[] {
+        return this.getEffectValues(EffectName.GainAbility).filter((ability) => ability.abilityType === abilityType);
```

Comment:
Does this cover all possible gained abilities? i.e. triggered, reactions, constants?

Reply:
Yeah you can retrieve any type using the combination of generic constraint and ability type, though you need to make sure that the two match.